strings/isisogram.go

```diff
@@ -0,0 +1,41 @@
+// Checks if a given string is an isogram.
+// An isogram is a word, phrase, or sentence in which no letter of the alphabet occurs more than once.
+// wiki: https://en.wikipedia.org/wiki/Heterogram_(literature)#Isograms
+// Author: M3talM0nk3y
+
+package strings
+
+import (
+	"regexp"
+	"strings"
+)
+
+func hasDigit(text string) bool {
+	re := regexp.MustCompile(`\d`)
+	return re.MatchString(text)
+}
+
+func hasSymbol(text string) bool {
+	re := regexp.MustCompile(`[-!@#$%^&*()+]`)
+	return re.MatchString(text)
+}
+
+func IsIsogram(text string) bool {
```

Comment:
What you have here is a first order isogram. The ideas of isograms is more general and there are first, second and third order isograms.

Can you generalise this to figure out first second and third order isograms?

Reply:
@tjgurwara99 Checking for first-order, second-order, and third-order isograms. Thanks for the feedback.